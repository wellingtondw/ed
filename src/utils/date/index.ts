type GetFullYearParams = {
  date: string | Date;
};
export const getFullYear = ({ date }: GetFullYearParams) => {
  return new Date(date).getFullYear();
};
