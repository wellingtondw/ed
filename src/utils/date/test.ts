import { getFullYear } from '.';

describe('utils/date', () => {
  it('should be able to get full year from date', () => {
    const dateString = '2022-05-04';
    let result = getFullYear({ date: dateString });

    expect(result).toBe(2022);

    const date = 'Wed Oct 18 2017 12:41:34 GMT+0000 (UTC)';
    result = getFullYear({ date });

    expect(result).toBe(2017);
  });
});
