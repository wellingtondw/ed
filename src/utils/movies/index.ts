import { MovieCardListDTO } from 'components/MovieCardList';

export type MoviesDTO = {
  poster_path: string;
  title: string;
  release_date: string;
  vote_average: number;
  id: number;
};

type FormatMoviesListParams = {
  items: MoviesDTO[];
};

export const formatMoviesList = ({ items }: FormatMoviesListParams) => {
  return items.map((item): MovieCardListDTO => {
    const { poster_path, title, release_date, vote_average, id } = item;

    return {
      posterImage: poster_path,
      title,
      date: release_date,
      rating: vote_average,
      id
    };
  });
};
