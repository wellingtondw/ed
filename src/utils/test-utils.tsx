import { createStore } from 'redux';
import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import { ThemeProvider } from 'styled-components';
import theme from 'styles/theme';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import rootReducer from '../store/modules/rootReducer';

const memoryHistory = createMemoryHistory();

const customRender = (
  ui: ReactElement,
  initialState = {},
  store = createStore(rootReducer, initialState),
  history = memoryHistory,
  { ...renderOptions }: RenderOptions = {}
) =>
  render(
    <Provider store={store}>
      <Router navigator={history} location={history.location}>
        <ThemeProvider theme={theme}>{ui}</ThemeProvider>
      </Router>
    </Provider>,
    renderOptions
  );

export * from '@testing-library/react';
export { customRender as render };
