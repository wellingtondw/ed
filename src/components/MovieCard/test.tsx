import { getFullYear } from '../../utils/date';
import { render, screen } from '../../utils/test-utils';

import { MovieCard } from '.';
import movieMock from './mock';

const { poster_path, title, release_date, vote_average, id } = movieMock;

type SutProps = {
  to?: string;
  loading?: boolean;
};
const sut = ({ to = '/' }: SutProps) => {
  return render(
    <MovieCard
      posterImage={poster_path}
      title={title}
      rating={vote_average}
      date={release_date}
      to={to}
    />
  );
};

describe('<MovieCard>', () => {
  it('should be able to render correctly', () => {
    const { container } = sut({});

    const year = getFullYear({ date: release_date });

    expect(screen.getByRole('img', { name: title })).toHaveAttribute('src', poster_path);
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(year)).toBeInTheDocument();
    expect(screen.getByText(vote_average)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should be able to redirect', async () => {
    const to = `/${id}`;
    sut({ to });

    const movieCardEl = screen.getByRole('link');

    expect(movieCardEl).toHaveAttribute('href', to);
  });
});
