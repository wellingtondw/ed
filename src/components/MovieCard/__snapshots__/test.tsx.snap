// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard> should be able to render correctly 1`] = `
.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 100%;
  max-width: 250px;
  border: 1px solid #EEE;
  border-radius: 0.4rem;
  cursor: pointer;
  background-color: #FFFFFF;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover img {
  -webkit-transform: scale3d(1.05,1.05,1.05);
  -ms-transform: scale3d(1.05,1.05,1.05);
  transform: scale3d(1.05,1.05,1.05);
}

.c1 {
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  width: 100%;
}

.c3 {
  width: 100%;
  padding: 0.8rem;
}

.c4 {
  font-size: 1.6rem;
  color: #29292E;
  font-weight: 700;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 4px;
}

.c6 {
  color: #835AFD;
  font-size: 1.4rem;
  font-weight: 700;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #835AFD;
  border-radius: 0.4rem;
  padding: 2px 4px;
  color: #FFFFFF;
}

.c7 svg {
  width: 1.2rem;
  height: 1.2rem;
  margin-right: 2px;
}

.c7 span {
  font-size: 1.2rem;
}

@media (max-width:768px) {
  .c0 {
    max-width: 180px;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media (max-width:768px) {
  .c6 {
    font-size: 1.2rem;
  }
}

<div>
  <a
    class="c0"
    href="/"
  >
    <div
      class="c1"
    >
      <img
        alt="Doutor Estranho no Multiverso da Loucura"
        class="c2"
        loading="lazy"
        src="/boIgXXUhw5O3oVkhXsE6SJZkmYo.jpg"
      />
    </div>
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        Doutor Estranho no Multiverso da Loucura
      </h2>
      <div
        class="c5"
      >
        <span
          class="c6"
        >
          2022
        </span>
        <div
          class="c7"
        >
          <svg
            aria-hidden="true"
            class="c8"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <span>
            7.5
          </span>
        </div>
      </div>
    </div>
  </a>
</div>
`;
